src/components/hooks/useAdaptiveCard.js

```diff
@@ -0,0 +1,36 @@
+import {useEffect, useContext, useState} from 'react';
+
+import {AdapterContext} from './contexts';
+
+/**
+ * An activity a person performs in Webex.
+ *
+ * @external Activity
+ * @see {@link https://github.com/webex/component-adapter-interfaces/blob/master/src/ActivitiesAdapter.js#L6}
+ */
+
+/**
+ * Custom hook that returns an AdaptiveCard data of the given ID.
+ *
+ * @param {string} cardActivityID  ID of the AdaptiveCard for which to return data.
+ * @returns {Activity} Data of the AdaptiveCard
+ */
+export default function useAdaptiveCard(cardActivityID) {
+  const [adaptiveCard, setAdaptiveCard] = useState({});
+  const {activitiesAdapter} = useContext(AdapterContext);
+
+  useEffect(() => {
+    const onError = (error) => {
+      throw error;
+    };
+    const subscription = activitiesAdapter.getActivity(cardActivityID)
+      .subscribe(setAdaptiveCard, onError);
+
+    return () => {
+      subscription.unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return JSON.parse(adaptiveCard);
```

Comment:
is this needed? I thought the content of the activity was already parsed into a javascript object @karinasigartau0798